src/store/sagas.js

```diff
@@ -0,0 +1,32 @@
+import { call, put, takeEvery, all } from 'redux-saga/effects'
+import {
+  saveNewDiarySuccess,
+  saveNewDiaryFailure,
+  clearDiaryFields
+} from './actions'
+import browserHistory from '../services/history'
+
+import { addDiary } from '../services/diaryApi'
+
+function* saveDiarySaga(action) {
+  const res = addDiary(action.payload)
+
+  // successful save
+  if (typeof res.id === 'string' && res.id.length > 0) {
+    yield put(saveNewDiarySuccess(res))
+    yield put(clearDiaryFields())
```

Comment:
So, every `put` does a dispatch.  Each dispatch can cause a lot of re-rendering or, at the very least, a lot of React reconciling.  

I've found it to be a good practice to try and only dispatch:

1. Actions directly caused by users
2. Actions caused by async actions completing (ex: successful fetch of data from server)

In this instance, I would have the reducer that deals with diary fields listen to the action created by `saveNewDiarySuccess` and clear the fields.

Reply:
That's good advice. What else have you found effective at limiting reconciliation/re-rendering? Do you still use re-select or some other form of memoization?